Eden/reqresp

eavivi4 opened the pull request:
I changed only the parts that used the phrases "MasterPort" or "SlavePort".

eavivi4 committed:
dev: Master/Slave terminology

Device.py, dma_device, gic_v3_its.hh
Gic.py, smmu_v3_ports, smmu_v3_slaveifc
Changes only to Master/SlavePorts

Change-Id: If2d271f8e0b4f410ca357a2e45c9b8aba18c59be

eavivi4 committed:
dev: Master/Slave terminology

SMMUv3.py, smmu_v3, I82094AA.py

Change-Id: I5a4f0d31599db80e3e68661b378fc1b66ad51edc

powerjg commented on src/dev/arm/SMMUv3.py:
```diff
@@ -74,10 +74,11 @@ class SMMUv3(ClockedObject):
+    master_walker = RequestPort(
         'Master port for SMMU initiated HWTW requests (optional)')
-    control = SlavePort('Control port for accessing memory-mapped registers')
+    control =
```
This is a bit of a strange way to solve the line length problem. I suggest doing it like the line above and putting `ResponsePort(` on this line and the "help" on the next line.

powerjg commented on src/dev/arm/smmu_v3.hh:
```diff
@@ -90,7 +90,7 @@ class SMMUv3 : public ClockedObject
     const MasterID masterId;
 
-    SMMUMasterPort    masterPort;
+    SMMUMasterPort    requestPort;
```
Do you think that "mainPort" would be a better name? Maybe we need to get input from Arm folks to know for sure.

The other ports have more descriptive names... I wonder why this one's name wasn't descriptive...

eavivi4 replied:
I don't really know what SMMUMasterPort does, so I don't think I know which one would be better. I think I changed that variable by accident because I thought it's related to MasterPort... Should I leave it as masterPort for now?

powerjg replied:
I'm not sure, tbh. Changing the variable names is definitely harder (and more controversial). That said, we need to do it at some point and get feedback on our names. However, I just have no idea how this is used, so maybe we can skip it for now :)

powerjg commented on src/dev/arm/smmu_v3_slaveifc.hh:
```diff
@@ -72,7 +72,7 @@ class SMMUv3SlaveInterface : public ClockedObject
     const Cycles mainTLBLat;
 
-    SMMUSlavePort *slavePort;
+    SMMUSlavePort *responsePort;
```
Similar to my comment above... I noticed you didn't change the ones below. Maybe this is a case where we should just ask :).

powerjg reviewed (commented):
Generally, this looks great!

I have a couple of comments below on the changes to member variable names. However, I don't have a strong opinion or know what the "right" solution is :). I'd love to hear what you think, though.

eavivi4 committed:
dev: Changes

Fixed new line spacing and
some files fixed back to master/slavePort
that are variables

Change-Id: Ib1fe8eadd62a19ac4485b8816f6c67e2feda491d

eavivi4 committed:
systemc: Master/Slave Terminology

TlmBridge.py, gem5_to_tlm.hh, tlm_to_gem5.hh

Change-Id: I84831a61a4aa1f5b967e393d667431a40dec8fb2

powerjg reviewed (commented):
LGTM!

eavivi4 committed:
mem: Master/Slave Terminology

src/mem/ruby files and under src/mem
addr_mapper.hh, AddrMapper.py, bridge, Bridge.py
coherent_xbar, comm_monitor.hh, CommMonitor.py

Change-Id: I669bc2f69cf8bc3e253bcddd783f549fff8eb155

powerjg commented on src/dev/arm/smmu_v3.cc:
```diff
@@ -310,7 +310,8 @@ SMMUv3::runProcessTiming(SMMUProcess *proc, PacketPtr pkt)
 
-            if (packetsToRetry.empty() && masterPort.sendTimingReq(action.pkt)) {
+            if (packetsToRetry.empty() &&
+            masterPort.sendTimingReq(action.pkt)) {
```
This should be tabbed one or two more times.

powerjg reviewed (commented):
One small thing below.

eavivi4 committed:
mem: Changes

When building, the previous commit gives an error
This commit is fixing that error to build

Change-Id: I343f208d3fb06484a49fe865ab5d947fbee56347

eavivi4 committed:
mem: Changes

Tab change

Change-Id: I2da82a056b63b7c066cf2f33829896cd42e5b373

powerjg reviewed (commented):
Still lookin' good!

eavivi4 committed:
mem: Master/Slave Terminology

ExternalMaster.py, MemChecker.py, MemDelay.py, SerialLink.py,
external_master.hh, mem_checker_monitor.hh,
packet_queue, serial_link

Change-Id: Idef013ae65d919c15a316aa6ad372784658d105c

eavivi4 committed:
mem: Master/Slave Terminology

XBar.py, Cache.py, base.hh

Change-Id: I3bc4e56f792dc4d010b415edb3da04aae1859142

powerjg commented on src/dev/arm/smmu_v3.cc:
```diff
@@ -310,7 +310,8 @@ SMMUv3::runProcessTiming(SMMUProcess *proc, PacketPtr pkt)
 
-            if (packetsToRetry.empty() && masterPort.sendTimingReq(action.pkt)) {
+            if (packetsToRetry.empty() &&
+                    masterPort.sendTimingReq(action.pkt)) {
```
There's nothing wrong with this change :). However, since we are no longer changing the masterPort name here, we should revert this to the original.
